core/templates/dev/head/domain/utilities/StatusObjectFactory.js

```diff
@@ -0,0 +1,32 @@
+// Copyright 2015 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Factory for creating new Status domain objects.
+ */
+
+oppia.factory('StatusObjectFactory', [function() {
+  var Status = function(reason, value) {
+    this.reason = reason;
```

Comment:
Please refer to https://github.com/oppia/oppia/blob/develop/core/templates/dev/head/domain/collection/CollectionObjectFactory.js for an example object factory. ``reason`` and ``value`` should be prefixed with underscores and we should have getters to retrieve the values.